src/population_loader.rs

```diff
@@ -1,112 +1,109 @@
 use crate::parameters_loader::Parameters;
 use ixa::{
     context::Context,
-    define_derived_property, define_person_property, define_person_property_with_default,
+    define_person_property, define_person_property_with_default,
     error::IxaError,
     global_properties::ContextGlobalPropertiesExt,
     people::{ContextPeopleExt, PersonId},
 };
 
 use serde::Deserialize;
+use std::path::PathBuf;
 
 #[derive(Deserialize, Debug)]
 #[allow(non_snake_case)]
-pub struct PeopleRecord {
+pub struct PeopleRecord<'a> {
     age: u8,
-    homeId: usize,
+    homeId: &'a [u8],
 }
 
 define_person_property!(Age, u8);
 define_person_property!(HomeId, usize);
 define_person_property_with_default!(Alive, bool, true);
 
-#[allow(clippy::cast_possible_truncation)]
-#[allow(clippy::cast_sign_loss)]
-const CENSUS_MAX: usize = 1e15 as usize;
-
-#[allow(clippy::cast_possible_truncation)]
-#[allow(clippy::cast_sign_loss)]
-const CENSUS_MIN: usize = 1e14 as usize;
-
-define_derived_property!(CensusTract, usize, [HomeId], |home_id| {
-    if (CENSUS_MIN..CENSUS_MAX).contains(&home_id) {
-        home_id / 10_000
-    } else {
-        0 //Err(IxaError::IxaError(String::from("Census tract invalid from homeId")))
-    }
-});
+define_person_property!(CensusTract, usize);
 
 fn create_person_from_record(
     context: &mut Context,
     person_record: &PeopleRecord,
 ) -> Result<PersonId, IxaError> {
-    let person_id =
-        context.add_person(((Age, person_record.age), (HomeId, person_record.homeId)))?;
+    let tract: usize = String::from_utf8(person_record.homeId[..11].to_owned())
```

Comment:
My first thought was to have something minimal in main without derived properties. My perception is that tract should be a group and not necessarily a person property. I suggest we let it be here and think about it as an issue since I think the 11 is very standard for fips codes.

Reply:
Thanks, this justification makes a ton of sense and sounds good. Look forward, I'd be curious what you think, then, about also having a more general group where people fit into households (a one-to-one group), and households fit into census tracts (a meta group that is also one-to-one)?